src/app/pages/app-bar/app-bar.component.ts

```diff
@@ -4,52 +4,48 @@ import { map } from 'rxjs/operators';
 import { ScrollDispatcher, CdkScrollable } from '@angular/cdk/scrolling';
 
 @Component({
-  selector: 'app-app-bar',
-  templateUrl: './app-bar.component.html',
-  styleUrls: ['./app-bar.component.scss']
+    selector: 'app-app-bar',
+    templateUrl: './app-bar.component.html',
+    styleUrls: ['./app-bar.component.scss'],
 })
 export class AppBarComponent implements OnInit {
-  @ViewChild('wrapper') wrapper: ElementRef;
-
-  list = presidents;
-  private readonly SHRINK_TOP_SCROLL_POSITION = 64;
-  shrinkToolbar = false;
-  scrollingSubscription: any;
-  scrollTop: any;
-  width: number;
-  constructor(
-    private scrollDispatcher: ScrollDispatcher,
-    private ngZone: NgZone
-  ) {}
-
-
-  ngOnInit() {
-    this.scrollDispatcher.scrolled()
-      .pipe(map((event: CdkScrollable) => this.getScrollPosition(event)))
-      .subscribe(scrollTop => this.ngZone.run(() => this.setScroll(scrollTop)));
-  }
-
-  ngAfterViewInit() {
-    if(this.wrapper) {
-      this.width = this.wrapper.nativeElement.clientWidth;
+    @ViewChild('wrapper') wrapper: ElementRef;
+
+    list = presidents;
```

Comment:
having the presidents list in a shared folder breaks the modularity of the examples (copy-pastability). I think each example should maintain (a copy of) their own data that they need.

Reply:
Removed the action list.